keanu-project/src/test/java/io/improbable/keanu/e2e/eight/EightSchools.java

```diff
@@ -0,0 +1,67 @@
+package io.improbable.keanu.e2e.eight;
+
+import io.improbable.keanu.algorithms.NetworkSamples;
+import io.improbable.keanu.algorithms.mcmc.NUTS;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.ConstantVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.HalfCauchyVertex;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Arrays;
+
+@Slf4j
+public class EightSchools {
+
+    public static void main(String[] args) {
+        EightSchools model = new EightSchools();
+        model.run();
+    }
+
+    public void run() {
+        KeanuRandom random = new KeanuRandom();
+
+        int J = 8;
```

Comment:
what's `J`?

Reply:
It's from the pymc example this comes from.